# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1347385726 14400
#      Tue Sep 11 13:48:46 2012 -0400
# Node ID 5f3a69a309a7ce816c6796305e3eff9ff29f871a
# Parent  9a8dbd6b6b20f31f6c5438393b26414a63f7b0b6
fix decl of mexPutVariable

* mexproto.h, mex.cc (mexPutVariable): Declare mxArray arg as const.
* mxarray.in.h, mex.cc (mxArray::as_octave_value (const mxArray*)):
Declare mxArray arg as const.

diff --git a/libinterp/interp-core/mex.cc b/libinterp/interp-core/mex.cc
--- a/libinterp/interp-core/mex.cc
+++ b/libinterp/interp-core/mex.cc
@@ -1889,17 +1889,17 @@ mxArray::~mxArray (void)
 void
 mxArray::set_name (const char *name_arg)
 {
   mxFree (name);
   name = strsave (name_arg);
 }
 
 octave_value
-mxArray::as_octave_value (mxArray *ptr)
+mxArray::as_octave_value (const mxArray *ptr)
 {
   return ptr ? ptr->as_octave_value () : octave_value (Matrix ());
 }
 
 octave_value
 mxArray::as_octave_value (void) const
 {
   return rep->as_octave_value ();
@@ -3209,17 +3209,17 @@ mexGetVariable (const char *space, const
 
 const mxArray *
 mexGetVariablePtr (const char *space, const char *name)
 {
   return mexGetVariable (space, name);
 }
 
 int
-mexPutVariable (const char *space, const char *name, mxArray *ptr)
+mexPutVariable (const char *space, const char *name, const mxArray *ptr)
 {
   if (! ptr)
     return 1;
 
   if (! name)
     return 1;
 
   if (name[0] == '\0')
diff --git a/libinterp/interp-core/mxarray.in.h b/libinterp/interp-core/mxarray.in.h
--- a/libinterp/interp-core/mxarray.in.h
+++ b/libinterp/interp-core/mxarray.in.h
@@ -481,17 +481,17 @@ public:
         mwSize sz =  sizeof (mxChar) * (strlen (str) + 1);
         retval = static_cast<char *> (mxArray::malloc (sz));
         strcpy (retval, str);
       }
 
     return retval;
   }
 
-  static octave_value as_octave_value (mxArray *ptr);
+  static octave_value as_octave_value (const mxArray *ptr);
 
 protected:
 
   octave_value as_octave_value (void) const;
 
 private:
 
   mutable mxArray_base *rep;
